src/game_api/savestate.hpp

```diff
@@ -1,19 +1,52 @@
 #pragma once
 
+#include <cstdint> // for uint32_t, int8_t
+
+#include "heap_base.hpp" // for HeapBase
+
 struct StateMemory;
+struct PRNG;
 
 class SaveState
 {
   public:
     /// Create a new temporary SaveState/clone of the main level state. Unlike save_state slots that are preallocated by the game anyway, these will use 32MiB a pop and aren't freed automatically, so make sure to clear them or reuse the same one to save memory. The garbage collector will eventually clear the SaveStates you don't have a handle to any more though.
-    SaveState();
+    SaveState()
+        : base(reinterpret_cast<uintptr_t>(malloc(8ull * 0x400000)))
+    {
+        save();
+    }
+    /// NoDoc
+    SaveState(uint8_t index)
+        : base(HeapBase::get(index)){};
     ~SaveState()
     {
         clear();
     }
+    /// Get the pre-allocated by the game save slot 1-4. Call as `SaveState.get(slot)`
```

Comment:
Should i change it so it's `SaveState:get(slot)`?
as i don't think we have any functions called with just the `.` dot

Reply:
ImGuiIo does use the dot function call, though idk if it's generated well in `spel2.lua`

Reply:
spel2.lua generates `function SaveState:get(save_slot) end` so i guess i should change this
For `modifierdown` it generates the standard format, so probably suggests `:`, dunno if there is a way to suggest a dot, i don't use it myself

Reply:
it's an issue for another time, we already have things in the API with the same problem, so I think it would be better to solve this in a specific PR for this

Reply:
k